src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexReadOperation.Sharding.cs

```diff
@@ -19,12 +22,26 @@ public partial class IndexReadOperation
 
         foreach (var field in query.Metadata.OrderBy)
         {
-            // https://issues.hibernatingrhinos.com/issue/RavenDB-17888
-            DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Grisha, DevelopmentHelper.Severity.Normal,
-                "make it work without storing the fields in the index");
+            string fieldName;
+            object fieldValue;
 
-            var fieldValue = document.GetField(field.Name.Value).StringValue(_state);
-            documentWithOrderByFields.AddOrderByField(field.Name.Value, fieldValue);
+            switch (field.OrderingType)
+            {
+                case OrderByFieldType.Long:
+                    fieldName = $"{field.Name}{Constants.Documents.Indexing.Fields.RangeFieldSuffixLong}";
```

Comment:
This is likely going to create a _lot_ of temporary variables, no? Once for each value in the query

Reply:
done